my-drosera-trap/trap-foundry-template/src/interfaces/ITrap.sol

```diff
@@ -0,0 +1,7 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+interface ITrap {
+    function collect() external view returns (bytes memory);
+    function isValid(bytes[] calldata data) external view returns (bool, bytes memory);
```

Comment:
Update isValid to be pure.

Note: we should see if there is a good way to make this interface importable. Perhaps that means just having an "drosera-lib" repo that is meant to house the interface and other libraries contributed by the community.